### PR TITLE
Populate funding rate inside offer in the maker

### PR DESCRIPTION
**Pull request by klochowicz**
Hard-code "0.0" until UI creates means for choosing funding rate.

Partial fix for #244

### CODE REVIEW

File: daemon/src/model.rs [hunk 1]
```diff
@@ -603,7 +603,7 @@ impl Timestamp {
 }
 
 /// Funding rate per SETTLEMENT_INTERVAL
-#[derive(Debug, Clone, Copy, PartialEq)]
+#[derive(Debug, Clone, Copy, PartialEq, Serialize, Deserialize)]
```
Comment (thomaseizinger) [inline daemon/src/model.rs:606]: Simply deriving these will allow you to side-step the invariants you are checking in the constructor. Deriving `Serialize` / `Deserialize` for a new-type is almost always incorrect :)

File: daemon/src/lib.rs [hunk 2]
```diff
@@ -238,13 +239,15 @@ where
         min_quantity: Usd,
         max_quantity: Usd,
         fee_rate: Option<u32>,
+        funding_rate: FundingRate,
```
Comment (thomaseizinger) [inline daemon/src/lib.rs]: Can we make this optional here? That would make this a backwards-compatible API change.

**Review by thomaseizinger — CHANGES_REQUESTED** [review]
Some comments!

File: daemon/src/lib.rs (hunk 2 above)
Comment (klochowicz) [reply daemon/src/lib.rs]: sure!

File: daemon/src/model.rs (hunk 1 above)
Comment (klochowicz) [reply daemon/src/model.rs:606]: please bear in mind that the invariants are temporary, until we support charging the maker. Happy to hear how would I go about it if I was planning to have long-term invariants in a type! Custom serialise/deserialise with fallbacking to an option if the invariants are not upheld?

Comment (thomaseizinger) [reply daemon/src/model.rs:606]: > fallbacking to an option if the invariants are not upheld?

No, you can just fail the deserialization with an arbitrary message! Usually, you'd just call the constructor to make sure you don't duplicate them and then just do `map_err(D::Error::custom)?;` to forward the error.

**Review by thomaseizinger — APPROVED** [review]

**Pull request merged**